### ISSUE TITLE
Even more false positives with `no-array-prototype-extensions` rule

### ISSUE THREAD

**Issue by simonihmig** [ember-cli/eslint-plugin-ember#1561]
On top of #1534, #1537 the rule also (IMO falsely) triggers when using methods of EmberArray, but not *implicitly* through prototype extensions (which is what this rule is trying to avoid AFAIU), but also when the EmberArray has been created *explicitly*, like in these cases:
* `A([...]).uniq()`
* `blog.comments.toArray()` (when `blog` is an ember-data model with a `hasMany` relationship - which also uses the Enumerable mixin)

While those cases maybe should be avoided in new code, they are not in any way deprecated (AFAIK), and as said above do not relate to prototype extensions, so they seem to be false positives here!

**Comment by bmish** [comment]
Just to spell out one potential heuristic we could add for this related example:

```js
async _fetchCategories(query) {
  const response = await this.store.query('connect/category', query);
  return response.toArray(); // false positive
}
```

If a variable initialization comes from `this.store`, then we could assume it's Ember Data related and ignore it.
